pkg/apis/apps/validation/validation_test.go

```diff
@@ -2112,9 +2112,10 @@ func TestValidateDaemonSetUpdate(t *testing.T) {
 	}
 
 	type dsUpdateTest struct {
-		old            apps.DaemonSet
-		update         apps.DaemonSet
-		expectedErrNum int
+		old                             apps.DaemonSet
+		update                          apps.DaemonSet
+		expectedErrNum                  int
+		enableSkipReadOnlyValidationGCE bool
```

Comment:
Good tests for DaemonSet - shouldn't we have the equivalent for Deployment and ReplicaSet ?

Reply:
I only see the replicaset.

Reply:
TestValidateDeployment() ?

Reply:
I try it.